common/robotPositioner.h

```diff
@@ -0,0 +1,186 @@
+    millimeter_t m_pos_Z;                                           // robot's z position
+    millimeter_t m_goalPositionX;                                   // goal position x
+    millimeter_t m_goalPositionY;                                   // goal position y
+    millimeter_t m_goalPositionZ;                                   // goal position z
+    millimeter_t m_distanceFromGoal;                                // Euclidean distance from goal coordinate
+    degree_t m_heading;                                             // heading (angle) of the robot
+    degree_t m_goalAngle;                                           // angle to turn after finding the correct location
+    degree_t m_bearingFinal;                                        // heading converted to range <-180, 180> from <0, 360>
+    degree_t m_bearingFromGoal;                                     // bearing (angle) from goal coordinate
+
+    millimeter_t m_robot_r;                                         // robot's wheel radius
+    millimeter_t m_robot_D;                                         // robots' axis length
+    millimeter_t m_stopping_distance;                               // if the robot's distance from goal < stopping dist, robot stops
+    degree_t m_allowed_heading_error;                               // the amount of error allowed in the final heading
+    double m_k1;                                                    // curveness of the path to the goal
+    double m_k2;                                                    // speed of turning on the curves
+    double m_alpha;                                                 // causes more sharply peaked curves 
+    double m_beta;                                                  // causes to drop velocity if 'k'(curveness) increases
+    meters_per_second_t m_max_velocity;                             // max velocity
+
+    // robot superclass
+    BoBRobotics::Robots::Tank &m_bot;   
+  
+
+    
+    void driveRobotWithVelocities(const meters_per_second_t v,
+                                  const degrees_per_second_t w,
+                                  const millimeter_t robot_r,
+                                  const millimeter_t robot_d) {
+        // v = r * (Vl+Vr)/2
+        // w = r * (Vr-Vl)/D
+        double a =  (robot_r/2).value(); 
+        double b =  (robot_r/2).value();
+        double c = ( robot_r / robot_d).value();
+        double d = (-(robot_r / robot_d).value());
+
+        double e=v.value(); 
+        double f=w.value();
+        double Vl,Vr;
+
+        // determinant
+        double det = (a*d - b*c);
+    
+        Vl = ((e*d - b*f)/det);
+        Vr = ((a*f - e*c)/det);
+        
+        // drive robot 
+        m_bot.tank(Vl, Vr);
+    }
+
+    
+    degree_t angleWrapAround(degree_t angle) {
+        // changing from <0,360> to <-180, 180>
+        if (angle <= degree_t(0)) { angle += degree_t(360); } 
+        if (angle > degree_t(180)) { angle -= degree_t(360); }
```

Comment:
Just an observation for future reference: you can use the unit suffixes here. So you can write ``0_deg`` ``360_deg`` etc. But your way is also fine.

Reply:
This function could also be marked ``static``...